# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50  

# Effort Controllers ---------------------------------------
joint1_F_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_F
joint1_B_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_B
joint1_L_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_L
joint1_R_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_R
joint2_F_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_F
joint2_B_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_B
joint2_L_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_L
joint2_R_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_R
joint3_F_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_F
joint3_B_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_B
joint3_L_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_L
joint3_R_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_R
joint4_F_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_F
joint4_B_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_B
joint4_L_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_L
joint4_R_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_R

# for 45 degree leg
joint1_LF_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_LF
joint1_LB_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_LB
joint1_RF_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_RF
joint1_RB_controller:
  type: effort_controllers/JointEffortController
  joint: joint1_RB
joint2_LF_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_LF
joint2_LB_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_LB
joint2_RF_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_RF
joint2_RB_controller:
  type: effort_controllers/JointEffortController
  joint: joint2_RB
joint3_LF_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_LF
joint3_LB_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_LB
joint3_RF_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_RF
joint3_RB_controller:
  type: effort_controllers/JointEffortController
  joint: joint3_RB
joint4_LF_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_LF
joint4_LB_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_LB
joint4_RF_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_RF
joint4_RB_controller:
  type: effort_controllers/JointEffortController
  joint: joint4_RB